crates/ark/src/fixtures/dummy_frontend.rs

```diff
@@ -42,19 +42,25 @@ impl DummyArkFrontend {
         let frontend = DummyFrontend::new();
         let connection_file = frontend.get_connection_file();
 
+        // Start the kernel in this thread so that panics are propagated
+        crate::start::start_kernel(
+            connection_file,
+            vec![
+                String::from("--interactive"),
+                String::from("--vanilla"),
+                String::from("--no-save"),
+                String::from("--no-restore"),
+            ],
+            None,
+            SessionMode::Console,
+            false,
+        );
+
+        // Start the REPL in a background thread, does not return and is never joined
         stdext::spawn!("dummy_kernel", || {
-            crate::start::start_kernel(
-                connection_file,
-                vec![String::from("--no-save"), String::from("--no-restore")],
-                None,
-                SessionMode::Console,
-                false,
-            );
+            RMain::start();
```

Comment:
I am mildly concerned about the fact that `R_INIT` sets "set" in `setup()` at the very end.

But we typically think of `R_INIT` as "R is ready to receive input", but that really isn't true because we haven't started the REPL yet (not until we call `start()` here).

I feel like its probably ok? Maybe `R_INIT` should be `R_STARTING` or something, I'm not sure. And the remaining helper would be `is_r_starting()`. If you look at the 1 place `is_r_initialized()` is currently being used, it is to capture output that is part of the startup banner. Switching the name to `is_r_starting()` seems like it would still make sense there.

Reply:
I was wondering about that too, but I couldn't move it to the `start()` method.

But I think that's ok because R _is_ ready to receive input or be accessed with well protected (top-level exec) API calls.